### PR TITLE
Feat(mysql): improve support for unsigned int types

### LINKED ISSUE #2166: Support MySQL unsigned integer
MySQL supports unsigned integer data type.
This can occur in two forms, with a parameterized int and without. 
```python
import sqlglot

sql = "CREATE TABLE t (id INT UNSIGNED)"
sql_parameterized = "CREATE TABLE t (id INT(10) UNSIGNED)"
sqlglot.parse_one(sql, read="mysql")
sqlglot
```
Now there is an exception in the library when parsing such a syntax.
Related discussion #2164 

**Official Documentation**
https://dev.mysql.com/doc/refman/8.0/en/numeric-type-syntax.html

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Fixes #2166

Reference: https://dev.mysql.com/doc/refman/8.0/en/numeric-type-syntax.html

cc: @Nitrino

### CODE REVIEW

File: sqlglot/dialects/mysql.py [hunk 1]
```diff
@@ -580,6 +587,21 @@ class Generator(generator.Generator):
             exp.DataType.Type.VARCHAR: "CHAR",
         }
 
+        UNSIGNED_TYPES = {
+            exp.DataType.Type.UBIGINT,
+            exp.DataType.Type.UINT,
+            exp.DataType.Type.UMEDIUMINT,
+            exp.DataType.Type.USMALLINT,
+            exp.DataType.Type.UTINYINT,
+        }
```
Comment (GeorgeSittas) [inline sqlglot/dialects/mysql.py:596]: Let me see if I can dry this up a bit with a dict that I can reuse in `TYPE_MAPPING`...

File: sqlglot/parser.py [hunk 2]
```diff
@@ -3359,6 +3368,13 @@ def _parse_types(
             self._retreat(index2)
 
         if not this:
+            if self._match_text_seq("UNSIGNED"):
+                unsigned_type_token = self.SIGNED_TO_UNSIGNED_TYPE_TOKEN.get(type_token)
+                if not unsigned_type_token:
+                    self.raise_error(f"Cannot convert {type_token.value} to unsigned.")
+
+                type_token = unsigned_type_token
```
Comment (tobymao) [inline sqlglot/parser.py:3376]: type_token = unsigned_type_token or type_token

File: tests/dialects/test_mysql.py [hunk 3]
```diff
@@ -6,6 +6,12 @@ class TestMySQL(Validator):
     dialect = "mysql"
 
     def test_ddl(self):
+        int_types = {"BIGINT", "INT", "MEDIUMINT", "SMALLINT", "TINYINT"}
+
+        for t in int_types:
+            self.validate_identity(f"CREATE TABLE t (id {t} UNSIGNED)")
+            self.validate_identity(f"CREATE TABLE t (id {t}(10) UNSIGNED)")
```
Comment (tobymao) [inline tests/dialects/test_mysql.py:13]: test a transpilation

**Review by tobymao — APPROVED** [review]

**Comment by Nitrino** [comment]
@GeorgeSittas Thanks a lot!

**Pull request merged**